File: classroomApplication/src/main/java/hu/elte/alkfejl/classroomApplication/model/Reservation.java
package hu.elte.alkfejl.classroomApplication.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.util.Date;

@Entity
@Table(name = "RESERVATIONS")
@Data
@AllArgsConstructor
@NoArgsConstructor
@EqualsAndHashCode
public class Reservation extends BaseEntity {

    @JoinColumn
    @ManyToOne(targetEntity = Room.class)
    private Room room;

    private Day day;

    private Date startTime;

    private Date endTime;

    public enum Day {
        VASARNAP, HETFO, KEDD, SZERDA, CSUTORTOK, PENTEK, SZOMBAT;
    }
}

File: classroomApplication/classroomApplication/src/app/model/Reservation.ts
import {Room} from './Room';

export class Day {
    static VASARNAP: String = 'VASARNAP';
    static HETFO: String = 'HETFO';
    static KEDD: String = 'KEDD';
    static SZERDA: String = 'SZERDA';
    static CSUTORTOK: String = 'CSUTORTOK';
    static PENTEK: String = 'PENTEK';
    static SZOMBAT: String = 'SZOMBAT';
}

export class Reservation {
    id: number;
    room: Room;
    //room: String
    day: Day;
    startTime: Date;
    endTime: Date;

    constructor(room?: Room, startTime?: Date, endTime?: Date, id?: number){
        this.room = room;
        this.startTime = startTime;
        this.endTime = endTime;
        this.id = id;
    }
}
